dev/bots/run_command.dart

```diff
@@ -83,12 +83,15 @@ Future<void> runCommand(String executable, List<String> arguments, {
   bool expectNonZeroExit = false,
   int expectedExitCode,
   String failureMessage,
-  bool printOutput = true,
+  OutputMode outputMode = OutputMode.print,
+  CapturedOutput output,
   bool skip = false,
   bool expectFlaky = false,
   Duration timeout = _kLongTimeout,
   bool Function(String) removeLine,
 }) async {
+  assert((outputMode == OutputMode.capture) == (output != null));
```

Comment:
Please add a message for this - e.g. `assert((outputMode == OutputMode.capture) == (output != null), 'The output parameter must not be null if outputMode == OutputMode.capture, and must be null otherwise')`.

Alternatively, perhaps this should be some kind of class itself rather than an enum and a callback...

Reply:
Fixed.  I think that keeping the enum and assert is a bit simpler for now.